src/decode.rs

```diff
@@ -5045,42 +5045,40 @@ pub unsafe fn rav1d_submit_frame(c: &mut Rav1dContext) -> Rav1dResult {
         }
     }
 
-    if (*f.dsp).ipred.intra_pred[DC_PRED as usize].is_none() {
```

Comment:
Does this now re-init on each frame? Is that intended?

Reply:
Hmm, no.  I misread that as the opposite.  Good catch.

Reply:
In that case, it's probably relying on it being zero-initialized somewhere before, which is not great.  Let me see if there's somewhere we can non-lazily initialize it.

Reply:
Is `Rav1dIntraPredDSPContext` public? We could just add a boolean `initialized` and check that?

Reply:
No, it's not `pub`.  That'll probably work.  I just want to check where it's actually getting initialized since it's happening implicitly and I want to make sure we avoid UB.

Reply:
I changed it to use your `initialized: bool` field suggestion.